src/main/java/org/folio/circulation/resources/ItemsInTransitResource.java

```diff
@@ -40,21 +61,141 @@ private void getMany(RoutingContext routingContext) {
+    MultipleRecords<Loan> loans) {
+
+    return of(() ->
+      multipleRequests.stream()
+        .map(itemAndRelatedRecord -> matchLoansToRequest(itemAndRelatedRecord, loans))
+        .collect(Collectors.toList()));
+  }
+
+  private ItemAndRelatedRecords matchLoansToRequest(
+    ItemAndRelatedRecords request,
+    MultipleRecords<Loan> loans) {
+
+    final Map<String, Loan> loanMap = loans.toMap(Loan::getItemId);
+    request
+      .setLoan(loanMap.getOrDefault(request.getItem().getItemId(), null));
+    return request;
+  }
+
+  private Result<MultipleRecords<Loan>> mapResponseToLoans(Response response) {
+    return MultipleRecords.from(response, Loan::from, "loans");
+  }
+
+  public CompletableFuture<Result<List<ItemAndRelatedRecords>>> fetchItemRelatedRecords(ResultItemContext resultItemContext,
+                                                                                        ItemRepository itemRepository,
+                                                                                        ServicePointRepository servicePointRepository) {
+    List<ItemAndRelatedRecords> itemAndRelatedRecords = resultItemContext.getResultListOfItems().stream()
+      .flatMap(records -> records.value().getRecords()
+        .stream()).map(item -> fetchRelatedRecords(itemRepository, servicePointRepository, item))
+      .map(CompletableFuture::join)
+      .map(item -> new ItemAndRelatedRecords(item.value()))
+      .collect(Collectors.toList());
+
+    return CompletableFuture.completedFuture(Result.succeeded(itemAndRelatedRecords));
+  }
+
+  private CompletableFuture<Result<Item>> fetchRelatedRecords(ItemRepository itemRepository,
+                                                              ServicePointRepository servicePointRepository,
+                                                              Item item) {
+    return CompletableFuture.completedFuture(Result.succeeded(item))
+      .thenComposeAsync(itemRepository::fetchItemRelatedRecords)
+      .thenComposeAsync(result -> result
+        .combineAfter(it -> servicePointRepository
+          .getServicePointById(it.getInTransitDestinationServicePointId()), Item::updateDestinationServicePoint));
+  }
+
+  private CompletableFuture<Result<List<ItemAndRelatedRecords>>> findRequestsByItemsIds(RequestRepository requestRepository,
+                                                                                        List<ItemAndRelatedRecords> batchItemIds) {
+    List<Result<MultipleRecords<Request>>> requestList = batchItemIds.stream()
+      .map(itemAndRelatedRecords -> {
+        final Result<CqlQuery> statusQuery = exactMatchAny("status", RequestStatus.openStates());
+        final Result<CqlQuery> itemIdsQuery = exactMatch("itemId", itemAndRelatedRecords.getItem().getItemId());
+
+        Result<CqlQuery> cqlQueryResult = statusQuery.combine(itemIdsQuery, CqlQuery::and);
+
+        return requestRepository.findByWithRelatedRecords(cqlQueryResult, OPEN_REQUEST_LIMIT);
```

Comment:
Does it make sense to search requests for several itemIds at the same time, not one by one?
to reduce number of api calls

Reply:
Done